### PR TITLE
added sv_race_server for race maps & @timgame only reload servers when changing tab

### PR DESCRIPTION
**Pull request by timgame**
Race feature will be used for the new race servers and contains commits:
18d82fa, 7156449 and a22a085

Remove useless variables and only reload servers when changing tab in ingame browser contains commits:
8983b29

### CODE REVIEW

**Comment by heinrich5991** [comment]
Adding new config options apart from `sv_race_server` is probably unnecessary.

This changes the behavior to not remove the projectiles on player death for the race gametype as far as I can tell. This differs from the behavior in teerace, which is probably the only other still running race modification. Is this intentional? If so, why do you want to change that?

"only reload servers when changing tab" is unrelated to the rest of the changes. Could it be considered wanted behavior if the user can click the tab again to force a reload?

**Comment by timgame** [comment]
I added the other settings because I thought it could be useful for players who want to tune the settings for their raceservers.

The `Race` gametype doesn't kill the projectile on player kill, I recreated this for me it also seems like teerace hosts `Race` servers.

"only reload server when changing tab" changed only the behaviour of the ingame browser since this was change for the menu browser some time ago. Use the refresh button to reload servers. I also removed a ton of useless code there.

File: src/game/client/components/menus_ingame.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <iostream>
```
Comment (heinrich5991) [inline src/game/client/components/menus_ingame.cpp]: This shouldn't be here.

**Comment by heinrich5991** [comment]
> I added the other settings because I thought it could be useful for players who want to tune the settings for their raceservers.

I think that all configuration options we add should have a clear use case, and should otherwise be avoided if possible. Adding configuration options has two downsides:
1) We're stuck with them forever. Servers or maps might start relying on these.
2) We're providing a less consistent playing experience.

> The `Race` gametype doesn't kill the projectile on player kill, I recreated this for me it also seems like teerace hosts `Race` servers.

teerace hosts teerace server, which have the gametype "Race". As far as I know the behavior of them is to remove projectiles on player death.


Only reloading on tab switch sounds fine if that's the behaviour elsewhere as well. In future, you should try to split your pull request so each of them only implements one feature.

**Comment by timgame** [comment]
on teerace the grenade isnt killed, i tested this.

**Comment by heinrich5991** [comment]
It would be easier to review this if the pull request was split into the serverbrowser and sv_race_server part. If you want help with that, I can assist you.

As the interaction between `sv_solo_server` and `sv_race_server` isn't entirely clear, perhaps one could merge these variables by making `sv_solo_server 2` the race server setting.

It seems the previous code for tried to not reload the server browser in case the user just switched to settings and back, your version does not preserve that, resulting in more refreshs.

I'm still trying to understand the implications of the changes in the laser code.

File: src/game/server/entities/character.cpp [hunk 2]
```diff
@@ -885,8 +885,7 @@ void CCharacter::Die(int Killer, int Weapon)
 	if(Server()->IsRecording(m_pPlayer->GetCID()))
 		Server()->StopRecord(m_pPlayer->GetCID());
 
-	// we got to wait 0.5 secs before respawning
```
Comment (heinrich5991) [inline src/game/server/entities/character.cpp]: This comment and the one below should probably be restored.

**Comment by timgame** [comment]
I will open two seperate pull requests.

**Comment by timgame** [comment]
I'm going to open two seperate pull requests.

**Pull request closed without merging**